testpar/t_vfd.c

```diff
@@ -6250,6 +6255,68 @@ main(int argc, char **argv)
 
     test_vector_io(mpi_rank, mpi_size);
 
+#ifdef H5_HAVE_SUBFILING_VFD
+
+    if (mpi_rank == 0)
+        printf("\n --- TESTING SUBFILING VFD: environment variables set to empty --- \n");
+
+    subfiling_subfile_prefix_saved         = getenv("H5FD_SUBFILING_SUBFILE_PREFIX");
```

Comment:
I believe that you need to strdup these

Reply:
True, although in this case we probably don't even need to try to save the values. @glennsong09 I think we can update this to just have the setenv/unsetenv pairs.

Reply:
Okay, I removed the value saving code.